src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1552,5 +1552,64 @@ public void ChainTipAdvancesAt150Blocks_AlternativeChainPresented_RelevantBlockD
                 }
             }
         }
+
+        /// <summary>
+        /// Issue 40 @ CT advances after last checkpoint to height LC + MaxReorg + 10. New chain is presented with
+        /// fork point at LC + 5, chain is not accepted.
+        /// </summary>
+        [Fact]
+        public void ChainAdvancesAfterACheckpoint_NewChainIsPresentedWithForkPoint_ChainIsNotAccepted()
+        {
+            // Chain header tree setup. Initial chain has 5 headers and checkpoint at h5.
+            // Example: h1=h2=h3=h4=(h5).
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints().Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            const int checkpointHeight = 5;
+            var checkpoint = new CheckpointFixture(checkpointHeight, chainTip.Header);
+            ctx.SetupCheckpoints(checkpoint);
+
+            // Setup max reorg to 100.
+            const int maxReorg = 100;
+            ctx.ChainStateMock.Setup(x => x.MaxReorgLength).Returns(maxReorg);
```

Comment:
You also need to set up finalised block.

Reply:
Done